### DISCUSSION TITLE
Breadcrumb Support

### DISCUSSION THREAD

**Discussion by dbareis** [osmandapp/OsmAnd#19400] [Ideas]
I've been looking for ages for breadcrumb support, which logically might be an option under "Configure Screen", basically it is a visual indication on the map as to where you have been.  My old tom-tom had this and it makes it much easier to correctly orientate the map when you are lost or in an unfamiliar area.  My 2019 Ford Everest also has them (see picture), it would be nice if the "dots" showed direction and age but apart from that it works well.

So far I really like the look of this app and its configurability.

![Breadcrumbs](https://github.com/osmandapp/OsmAnd/assets/295230/07ed13ea-d646-40c6-9e95-53d2a23c85d5)

**Comment by tmusolf** [comment]
Have you checked out the [trip recording plugin](https://osmand.net/docs/user/plugins/trip-recording)?  That's what I use to record and display where I've been. This track can also be exported for use outside of OSMAnd.

**Reply by dbareis** [reply]
No I haven't but I don't want to record anyway, I want to see it on the display along with the map at all times.  I have currently gone off using OsmAnd due to the very poor Android Auto support.

**Reply by tommycrock** [reply]
FWIW StreetComplete does this sort of thing although it uses a fat blue light transparent trail. (I say this simply to give another example of the functionality that I think is being requested)
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/5749884/5b839ac8-2e4c-45b3-b80c-911df3c55852"  width="300" height="auto"/> </td>
  </tr>
</table>

**Reply by sjvudp** [reply]
To display breadcrumbs you must record the track; may be not as GPX, but in memory; anyway you need to record it. You can only display the data you have.

**Reply by dbareis** [reply]
> To display breadcrumbs you must record the track; may be not as GPX, but in memory; anyway you need to record it. You can only display the data you have.

Not true, the app knows where I have been, My 10 year old TomTom did this.  The first Android app that I find doing this is the app I will probably use .

**Reply by tommycrock** [reply]
Yes, the system needs to somehow record where you have been and when to implement breadcrumbs - hardly a big hurdle for osmand - but from a user's point of view it's very different from recording trips/tracks. Tracks should persist, be exportable and relate to user command to start/pause/stop etc. They should also continue recording indefinitely when the app is closed, which would probably be undesirable for breadcrumbs. Breadcrumbs should age and disappear and their recording should be automatic (once set).
